services/api/src/submission/checks/base-checks/color-contrast-greater-than-or-equal.check.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable, Logger } from "@nestjs/common";
+import ColorContrastChecker from "color-contrast-checker";
+import { ColorTranslator } from "colortranslator";
+import { DOMWindow } from "jsdom";
+
+import { CodeLevelSubmission as Submission } from "../../graphql/models/code-level-submission.model";
+import { Rule } from "../../interfaces/rule.interface";
+import { RuleCheckResult } from "../../interfaces/rule-check-result.interface";
+import { JsdomCheck } from "./jsdom-check";
+
+@Injectable()
+export class ColorContrastGreaterThanOrEqual extends JsdomCheck {
+  constructor(logger: Logger) {
+    super(logger);
+  }
+
+  public async evaluateRule(window: DOMWindow, submission: Submission, rule: Rule): Promise<RuleCheckResult> {
+    for (const option of ["selector", "minContrastValue", "property1", "property2"]) {
+      if (!this.ruleHasOption(rule, option)) {
+        return this.checkConfigurationError(submission, rule, option);
+      }
+    }
+
+    const { selector, minContrastValue, property1: cssProperty1, property2: cssProperty2 } = rule.options;
+    const matchingElements = window.document.querySelectorAll(selector);
+
+    let hasSufficientColorContrast = false;
+    const ccc = new ColorContrastChecker();
+    for (const element of matchingElements) {
+      const styles = window.getComputedStyle(element);
```

Comment:
Does `getComputedStyles` return the correct background color for an element if the color is specified on a parent element? For example in the following example:

```html
<div style="background-color: #000">
  <p style="color: #fff">Some text</p>
</div>
```

Reply:
No it doesn't because the background color is not inherited. It will return `rgba(0,0,0,0)` (transparent) instead which is the default background for most elements. 

Theoretically, we could recursively move up the tree if the background color is `rgba(0,0,0,0)` to find the real background. But I don't think that's necessary right now.

The more problematic behavior right now is that the ColorContrastChecker only takes hex values as input and the ColorTranslator ignores the alpha channel when converting to hex. Meaning `rgba(0,0,0,0)` is interpreted as black.

So I guess, there are a lot of edge cases here but I would ignore them for now since we only use this check for one simple check. Let's see if we use it for more complex examples in the future. Then we could think of building more custom logic.

Reply:
Ok. Then I'd say lets just be aware that there are edge cases and extend the check once we hit them 👍